include/kframework/foundry.md

```diff
@@ -552,6 +552,59 @@ This rule then takes the address using `#asWord(#range(ARGS, 0, 32))` and makes
       requires SELECTOR ==Int selector ( "symbolicStorage(address)" )
 ```
 
+#### `freshSInt` - Returns a single symbolic signed integer.
+
+```
+function freshSInt(uint8) external returns (uint256);
+```
+
+`foundry.call.freshSInt` will match when the `freshSInt` cheat code function is called.
+This rule returns a symbolic integer of up to the bit width that was sent as an argument.
+
+```k
+    rule [foundry.call.freshSInt]:
+         <k> #call_foundry SELECTOR ARGS => . ... </k>
+         <output> _ => #bufStrict(32, ?WORD) </output>
```

Comment:
This doesn't work because `#bufStrict(SIZE, DATA)` requires `0 <=Int DATA`. I don't know if it makes sense to have a signed version of this cheatcode, instead it might be better to just implement `freshSInt...()` like
```
function freshSInt256() external returns (int256) {
    return int256(kevm.freshUInt256());
}
```

Reply:
Right, nice catch
I don't know if there is less overhead with a native cheatcode ? It seems like there isn't any, we might defer to @ehildenb

Reply:
Native cheatcodes will usually have lower overhead, because they don't require executing any code aside from calling into the KEVMCheats contract.

Either approach seems fine.